src/components/reservation/reservation-modal.tsx

```diff
@@ -94,11 +98,23 @@ const ReservationModal = ({ manifestation }: ReservationModalProps) => {
       }
     };
     mutate(batch, {
-      onSuccess: () => setReservationDidSuccess(true),
-      onError: () => setReservationDidSuccess(false)
+      onSuccess: (res) => {
+        // this state is used to show the success or error modal
+        setReservationResponse(res);
+        // because after a successful reservation the holdings (reservations) are updated
+        queryClient.invalidateQueries(getGetHoldingsV3QueryKey());
+      }
     });
   };
 
+  const reservationResult = reservationResponse?.reservationResults
+    ? reservationResponse.reservationResults[0].result
+    : null;
+
+  const reservationDetails = reservationResponse?.reservationResults
+    ? reservationResponse.reservationResults[0].reservationDetails
+    : null;
```

Comment:
Consider if this can be simplified with more null chaining e.g. 
`reservationResult =reservationResponse?.reservationResults[0]?.reservationDetails;`